iterfzf/__init__.py

```diff
@@ -27,95 +26,110 @@ def iterfzf(
+    multi=False,
+    mouse=True,
+    print_query=False,
+    cycle=False,
     # Layout:
-    prompt='> ',
+    prompt="> ",
     preview=None,
     # Misc:
-    query='', encoding=None, executable=BUNDLED_EXECUTABLE or EXECUTABLE_NAME
+    query="",
+    encoding=None,
+    executable=BUNDLED_EXECUTABLE or EXECUTABLE_NAME,
 ):
-    cmd = [executable, '--no-sort', '--prompt=' + prompt]
+    cmd = [executable, "--no-sort", "--prompt=" + prompt]
     if not extended:
-        cmd.append('--no-extended')
+        cmd.append("--no-extended")
     if case_sensitive is not None:
-        cmd.append('+i' if case_sensitive else '-i')
+        cmd.append("+i" if case_sensitive else "-i")
     if exact:
-        cmd.append('--exact')
+        cmd.append("--exact")
+    if cycle:
+        cmd.append("--cycle")
     if multi:
-        cmd.append('--multi')
+        cmd.append("--multi")
     if not mouse:
-        cmd.append('--no-mouse')
-    if print_query:
-        cmd.append('--print-query')
+        cmd.append("--no-mouse")
     if query:
-        cmd.append('--query=' + query)
+        cmd.append("--query=" + query)
     if preview:
-        cmd.append('--preview=' + preview)
+        cmd.append("--preview=" + preview)
+    # always use --print-query: makes things simpler later as return values are always the same
+    cmd.append("--print-query")
+    query_results = _exec_cmd(cmd, iterable, encoding, executable)
+    query, results = query_results[0], query_results[1:]
+    if len(results) > 1 and results[0] is not None:
+        return_value = results if multi else results[0]
+    else:
+        return_value = None
+    if print_query:
+        return query, return_value
+    else:
+        return return_value
+
+
+def _exec_cmd(cmd, iterable, encoding, executable):
+    """Create subprocess with fzf and return its output to the caller.
+    Most of the work here is related to making sure that the encodings
+    in both directions are correct.
+    In either case returns a tuple where the first element is the query
+    which will be returned if query_result=True as well as the chosen items
+
+    Arguments:
+        cmd List[str] -- Command to be executed via subprocess
+        ... -- just passed through from function call above
+
+    Raises:
+        ValueError: If iterable contains unsuitable elements
+
+    Returns:
+        str,List[str] -- The query (search term entered in fzf) + list of results
+    """
     encoding = encoding or sys.getdefaultencoding()
     proc = None
-    stdin = None
-    byte = None
-    lf = u'\n'
-    cr = u'\r'
-    for line in iterable:
-        if byte is None:
-            byte = isinstance(line, bytes)
-            if byte:
-                lf = b'\n'
-                cr = b'\r'
-        elif isinstance(line, bytes) is not byte:
+    iterator = iter(iterable)
+    first = next(iterator, None)
+    if first is None:
+        return None, [None]  # early exit if iterable is empty
```

Comment:
The existing behavior on an empty iterable had not been early return, but showing an empty `fzf` window (i.e., `echo -n | fzf`).  IMHO the existing behavior is more predictable and should be maintained.